controllers/provider-alicloud/pkg/alicloud/client/client.go

```diff
@@ -151,3 +153,82 @@ func (c *storageClient) DeleteBucketIfExists(ctx context.Context, bucketName str
 func ComputeStorageEndpoint(region string) string {
 	return fmt.Sprintf("https://oss-%s.aliyuncs.com/", region)
 }
+
+type clientFactory struct {
```

Comment:
As can be seen here a new factory is added. This is because the old Factory and its FactoryFunc implementation is not good enough for multiple types of clients. Since we want to enable image sharing asap, we here first temporarily add a new factory interface, next step would be refactor client package code and merge the multiple client factories into one.